packages/network/src/logic/node/ProxyStreamConnectionManager.ts

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    processLeaveRequest(message: UnsubscribeRequest, nodeId: NodeId): void {
+        const { streamId, streamPartition } = message
+        const spid = new SPID(streamId, streamPartition)
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasInOnlyConnection(spid, nodeId)) {
+            this.removeConnection(spid, nodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, nodeId, spid)
+        }
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasOutOnlyConnection(spid, nodeId)) {
+            this.removeConnection(spid, nodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, nodeId, spid)
+            logger.info(`Proxy node ${nodeId} closed one-way stream connection for ${spid}`)
+        }
+    }
+
+    async processPublishStreamRequest(message: PublishStreamConnectionRequest, nodeId: string): Promise<void> {
+        const { streamId, streamPartition } = message
+        const spid = new SPID(streamId, streamPartition)
+
+        // More conditions could be added here, ie. a list of acceptable ids or max limit for number of one-way this
+        const isAccepted = this.streamManager.isSetUp(spid) && this.acceptProxyConnections
+        if (isAccepted) {
+            this.streamManager.addInOnlyNeighbor(spid, nodeId)
+        }
+        return await this.nodeToNode.respondToPublishOnlyStreamConnectionRequest(nodeId, spid, isAccepted)
+    }
+
+    processPublishStreamResponse(message: PublishStreamConnectionResponse, nodeId: string): void {
+        const { streamId, streamPartition, accepted } = message
+        const spid = new SPID(streamId, streamPartition)
+        if (accepted) {
+            this.getConnection(nodeId, spid)!.state = State.CONNECTED
+            this.streamManager.addOutOnlyNeighbor(spid, nodeId)
+            this.node.emit(Event.PUBLISH_STREAM_ACCEPTED, nodeId, spid)
+        } else {
+            this.removeConnection(spid, nodeId)
+            this.node.emit(Event.PUBLISH_STREAM_REJECTED, nodeId, spid, `Target node ${nodeId} rejected publish only stream connection ${spid.key}`)
+        }
+    }
+
+    async reconnect(targetNodeId: NodeId, spid: SPID): Promise<void> {
+        const connection = this.getConnection(targetNodeId, spid)!
+        if (connection.state !== State.RECONNECTING) {
+            connection.state = State.RECONNECTING
+        }
+        const trackerId = this.trackerManager.getTrackerId(spid)
+        const trackerAddress = this.trackerManager.getTrackerAddress(spid)
+        try {
+            await this.trackerManager.connectToSignallingOnlyTracker(trackerId, trackerAddress)
+            await promiseTimeout(this.nodeConnectTimeout, this.nodeToNode.connectToNode(targetNodeId, trackerId, false))
+            await this.nodeToNode.requestPublishOnlyStreamConnection(targetNodeId, spid)
+            logger.trace(`Successful proxy stream reconnection to ${targetNodeId}`)
```

Comment:
Some duplicated logic with `openOutgoingStreamConnection`. Generally when you have connection and re-connection logic, you can collapse the two into one thing.